### PR TITLE
WF-64209 Prepare and include the UG document for AutoGenerateColumnsModeForCustomType

### PR DESCRIPTION
**Pull request by ghost**
### Description ###
Provided UG documentation for AutoGenerateColumnsModeForCustomType in winforms-datagrid.

### CODE REVIEW

File: WindowsForms/DataGrid/Columns.md [hunk 1]
```diff
@@ -257,6 +257,59 @@ Keeps the columns which matches in the current data source and the data operatio
 </tr>
 </table>
 
+#### Auto generating columns for complex type
+
+Custom type (complex type) properties in data object can be auto-generated by setting [AutoGenerateColumnsForCustomType](https://help.syncfusion.com/cr/windowsforms/Syncfusion.WinForms.DataGrid.SfDataGrid.html#Syncfusion_WinForms_DataGrid_SfDataGrid_AutoGenerateColumnsForCustomType) property as true. Default value is false.
```
Comment (Chrispineagunja) [inline WindowsForms/DataGrid/Columns.md:262]: Custom type (complex type) properties in the data object can be auto-generated by setting the [AutoGenerateColumnsForCustomType](https://help.syncfusion.com/cr/windowsforms/Syncfusion.WinForms.DataGrid.SfDataGrid.html#Syncfusion_WinForms_DataGrid_SfDataGrid_AutoGenerateColumnsForCustomType) property as true. The default value is false.

File: WindowsForms/DataGrid/Columns.md [hunk 2]
```diff
@@ -257,6 +257,59 @@ Keeps the columns which matches in the current data source and the data operatio
...
+Custom type properties will be auto-generated through `AutoGenerateColumnsModeForCustomType` property.
```
Comment (Chrispineagunja) [inline WindowsForms/DataGrid/Columns.md:263]: Custom type properties will be auto-generated through the `AutoGenerateColumnsModeForCustomType` property.

File: WindowsForms/DataGrid/Columns.md [hunk 3]
```diff
@@ -257,6 +257,59 @@ Keeps the columns which matches in the current data source and the data operatio
...
+<th>
+Behavior
+</th>
+</tr>
+<tr>
+<td>
+<code>Child</code>
+</td>
+<td>
+Specifies that the columns for all inner properties of custom type column will be auto generated.
```
Comment (Chrispineagunja) [inline WindowsForms/DataGrid/Columns.md:292]: Specifies that the columns for all inner properties of the custom type column will be auto-generated.

File: WindowsForms/DataGrid/Columns.md [hunk 4]
```diff
@@ -257,6 +257,59 @@ Keeps the columns which matches in the current data source and the data operatio
...
+</td>
+<td>
+Specifies that the columns for both the custom type and its inner properties will be auto generated.
+</td>
+</tr>
+</table>
+
 #### Customize Auto-generated Columns
 
 The auto generated column can be customized by handling the [AutoGeneratingColumn](https://help.syncfusion.com/cr/windowsforms/Syncfusion.WinForms.DataGrid.SfDataGrid.html) event. `AutoGeneratingColumn` event occurs when the column is auto generated.
```
Comment (Chrispineagunja) [inline WindowsForms/DataGrid/Columns.md:315]: The auto-generated column can be customized by handling the [AutoGeneratingColumn](https://help.syncfusion.com/cr/windowsforms/Syncfusion.WinForms.DataGrid.SfDataGrid.html) event. The `AutoGeneratingColumn` event occurs when the column is auto-generated.

File: WindowsForms/DataGrid/Columns.md [hunk 5]
```diff
@@ -257,6 +257,59 @@ Keeps the columns which matches in the current data source and the data operatio
...
+<td>
+Specifies that the columns for all inner properties of custom type column will be auto generated. 
+</td>
+</tr>
+<tr>
+<td>
+<code>Parent</code>
+</td>
+<td>
+Specifies that the column for only the custom type will be auto generated.
```
Comment (Chrispineagunja) [inline WindowsForms/DataGrid/Columns.md:300]: Specifies that the column for only the custom type will be auto-generated.

File: WindowsForms/DataGrid/Columns.md (hunk 2 above)
Comment (sethuramkumar) [reply WindowsForms/DataGrid/Columns.md]: Hi @Chrispineagunja ,

I have modified the content as per the concern provided.

File: WindowsForms/DataGrid/Columns.md (hunk 3 above)
Comment (sethuramkumar) [reply WindowsForms/DataGrid/Columns.md]: Hi @Chrispineagunja ,

I have modified the content as per the concern provided.

File: WindowsForms/DataGrid/Columns.md (hunk 4 above)
Comment (sethuramkumar) [reply WindowsForms/DataGrid/Columns.md]: Hi @Chrispineagunja ,

I have modified the content as per the concern provided.

File: WindowsForms/DataGrid/Columns.md (hunk 5 above)
Comment (sethuramkumar) [reply WindowsForms/DataGrid/Columns.md]: Hi @Chrispineagunja ,

I have modified the content as per the concern provided.

**Review by susmitha-sundar — APPROVED** [review]

**Pull request merged**